src/Nethermind/Nethermind.Evm/Tracing/GethStyle/Javascript/CallFrame.cs

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Numerics;
+using Microsoft.ClearScript;
+using Microsoft.ClearScript.JavaScript;
+using Nethermind.Core;
+using Nethermind.Int256;
+
+namespace Nethermind.Evm.Tracing.GethStyle.Javascript;
+
+public record CallFrame
+{
+    public string? Type { get; set; }
+
+    public Address From
+    {
+        get => _from;
+        set
+        {
+            _from = value;
+            _fromConverted = null;
+        }
+    }
+
+    public Address To
+    {
+        get => _to;
+        set
+        {
+            _to = value;
+            _toConverted = null;
+        }
+    }
+
+    public ReadOnlyMemory<byte> Input
+    {
+        get => _input;
+        set
+        {
+            _input = value;
+            _inputConverted = null;
+        }
+    }
+
+    public long Gas { get; set; }
+
+    public UInt256? Value
+    {
+        get => _value;
+        set
+        {
+            _value = value;
+            _valueConverted = null;
+        }
+    }
+
+    private ITypedArray<byte>? _fromConverted;
+    private ITypedArray<byte>? _toConverted;
+    private ITypedArray<byte>? _inputConverted;
+    private dynamic? _valueConverted;
+    private Address _from;
+    private Address _to;
+    private ReadOnlyMemory<byte> _input;
+    private UInt256? _value;
+
+    // ReSharper disable InconsistentNaming
+    public string? getType() => Type;
```

Comment:
What's the reason for "camelCase" here? Kind of auto-mapping?

Reply:
yes we follow javascript convention here for auto-mapping through reflection

Reply:
You can find the API description here: https://geth.ethereum.org/docs/developers/evm-tracing/custom-tracer